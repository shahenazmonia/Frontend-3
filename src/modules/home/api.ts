import { message } from "antd";
import axios from "axios";
import { PaymentInput } from "./types";

const api = "http://178.63.13.157:8090/mock-api/api";

/**  all the requests should be handled by using redux or context to pass the data to the global 
 state in case we need it in different places and to manage all the requests/errors from one place
*/

export const getProjects = () => {
  return axios
    .get(`${api}/projects`)
    .then(({ data }) => data.data)
    .catch((error) => message.error(error));
};

export const getGateways = () => {
  return axios
    .get(`${api}/gateways`)
    .then(({ data }) => data.data)
    .catch((error) => message.error(error));
};

export const getUsers = () => {
  return axios
    .get(`${api}/users`)
    .then(({ data }) => {
      return data;
    })
    .catch((error) => message.error(error));
};

export const getPayments = (input: PaymentInput) => {
  return axios
    .post(`${api}/report`, { ...input })
    .then(({ data }) => data.data)
    .catch((error) => message.error(error));
};
