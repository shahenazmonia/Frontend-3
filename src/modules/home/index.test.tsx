import React from "react";
import { RenderResult, screen } from "@testing-library/react";
import { Home } from ".";
import * as API from "./api";

let wrapper: RenderResult;

test("Home component should be defined", () => {
  expect(Home).toBeDefined();
});

test("Home Shallow rendering", () => {
  expect(wrapper).toMatchSnapshot();
});

test("getProjects should be defined", () => {
  expect(API.getProjects).toBeDefined();
});
