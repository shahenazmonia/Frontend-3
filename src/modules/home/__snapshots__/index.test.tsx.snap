// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Shallow rendering 1`] = `undefined`;
